src/basis/type/enumeration.js

```diff
@@ -0,0 +1,44 @@
+function enumeration(values) {
+  if (!Array.isArray(values))
+  {
+    /** @cut */ basis.dev.warn('basis.type.enum constructor expected array but got ' + values + '. Wrapping into array');
+    values = [values];
+  }
+
+  if (!values.length)
+  {
+    /** @cut */ basis.dev.warn('basis.type.enum constructor expected non-empty array but got empty. Falling back to [null]');
+    values = [null];
```

Comment:
@lahmatiy Empty array is not ok, because in this case we can't assign any value to transform